warehouse/query-core/src/test/java/datawave/query/iterator/logic/AndIteratorTest.java

```diff
@@ -18,281 +23,433 @@ public void setup() {}
-        Assert.assertTrue(unsourcedIterator.isContextRequired());
-        Assert.assertFalse(iterator.hasNext());
+        AndIterator<String> iterator = new AndIterator<>(includes);
+        assertNoHits(iterator);
     }
     
+    // A && (!B && !C) ==> A && !(B || C)
     @Test
     public void testDeferredNegatedChild() {
         Set<NestedIterator<String>> includes = new HashSet<>();
         Set<NestedIterator<String>> excludes = new HashSet<>();
-        excludes.add(getItr(Lists.newArrayList("a", "b"), true));
-        excludes.add(getItr(Lists.newArrayList("b", "c"), true));
+        excludes.add(getIter("a", "b"));
+        excludes.add(getIter("b", "c"));
         
-        AndIterator unsourcedIterator = new AndIterator(includes, excludes);
+        AndIterator<String> unsourcedIterator = new AndIterator<>(includes, excludes);
         
         includes = new HashSet<>();
         includes.add(unsourcedIterator);
-        includes.add(getItr(Lists.newArrayList("c"), false));
+        includes.add(getIter("c"));
         
-        AndIterator iterator = new AndIterator(includes);
-        iterator.initialize();
-        
-        Assert.assertFalse(iterator.isContextRequired());
-        Assert.assertTrue(unsourcedIterator.isContextRequired());
-        Assert.assertFalse(iterator.hasNext());
+        AndIterator<String> iterator = new AndIterator<>(includes);
+        assertNoHits(iterator);
     }
     
     @Test
     public void testContextWhenNotRequiredMoveAll() {
         Set<NestedIterator<String>> includes = new HashSet<>();
-        includes.add(getItr(Lists.newArrayList("c", "d", "e"), false));
-        includes.add(getItr(Lists.newArrayList("c", "d", "e"), false));
-        
-        AndIterator iterator = new AndIterator(includes);
-        iterator.initialize();
+        includes.add(getIter("c", "d", "e"));
+        includes.add(getIter("c", "d", "e"));
         
-        Assert.assertFalse(iterator.isContextRequired());
-        iterator.setContext("e");
-        Assert.assertTrue(iterator.hasNext());
-        Assert.assertEquals("c", iterator.next());
-        Assert.assertTrue(iterator.hasNext());
-        Assert.assertEquals("e", iterator.next());
-        Assert.assertFalse(iterator.hasNext());
+        AndIterator<String> iterator = new AndIterator<>(includes);
+        assertHits(iterator, "c", "d", "e");
     }
     
     @Test
     public void testContextWhenNotRequiredShortCircuit() {
         Set<NestedIterator<String>> includes = new HashSet<>();
-        includes.add(getItr(Lists.newArrayList("c", "s", "z"), false));
-        includes.add(getItr(Lists.newArrayList("c", "z"), false));
+        includes.add(getIter("c", "s", "z"));
+        includes.add(getIter("c", "z"));
         
-        AndIterator iterator = new AndIterator(includes);
-        iterator.initialize();
-        
-        Assert.assertFalse(iterator.isContextRequired());
-        iterator.setContext("d");
```

Comment:
Is this no longer relevant for these tests?

Reply:
Setting the context is no longer required for the tests, that was a means to construct arbitrary scenarios without configuring a lot of input data.